File: go-service/models/branch.go
package models

import "time"

// Branch represents a physical branch/location
type Branch struct {
	ID        int        `db:"Id" json:"id"`
	Name      string     `db:"Name" json:"name"`
	Code      string     `db:"Code" json:"code"`
	Address   *string    `db:"Address" json:"address,omitempty"`
	City      *string    `db:"City" json:"city,omitempty"`
	Phone     *string    `db:"Phone" json:"phone,omitempty"`
	Email     *string    `db:"Email" json:"email,omitempty"`
	IsActive  bool       `db:"IsActive" json:"isActive"`
	CreatedAt time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

File: frontend/src/app/core/domain/branch/branch.ts
import { Inventory, User} from '../domain.barrel';

/**
 * Represents a Branch/Location
 */
export interface Branch {
  /** ID */
  id: number;

  /** Name - branch name */
  name: string;

  /** Code - branch code/identifier */
  code: string;

  /** Address - physical address */
  address?: string | null;

  /** City - city location */
  city?: string | null;

  /** Phone - contact phone */
  phone?: string | null;

  /** Email - contact email */
  email?: string | null;

  /** Is Active - is the branch active */
  isActive?: boolean;

  /** Users - users assigned to this branch */
  users?: User[];

  /** Inventory Records - inventory at this branch */
  inventoryRecords?: Inventory[]; // Changed from Inventory[] to any[] to avoid circular dependency

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
